set.mm

```diff
@@ -518224,6 +518263,512 @@ such that every nonzero prime ideal contains a prime element (this is a
+      VIVFCUTUSUQAVGCFACBMVGCFPVAJCFBVBVCVDVE $.
+  $}
+
+  ${
+    $d Q x y $.  $d R x y $.
+    rprmdvds.2 $e |- P = ( RPrime ` R ) $.
+    rprmdvds.3 $e |- U = ( Unit ` R ) $.
+    rprmdvds.5 $e |- ( ph -> R e. V ) $.
+    rprmdvds.6 $e |- ( ph -> Q e. P ) $.
+    $( A ring prime is not a unit.  (Contributed by Thierry Arnoux,
+       18-May-2025.) $)
+    rprmnunit $p |- ( ph -> -. Q e. U ) $=
+      ( vx vy cfv wcel wn cv wbr wral eqid simprbda c0g csn cun wceq eqidd crpm
+      cbs cdif eleqtrdi cmulr co cdsr wo wi isrprm syl2anc eldifbd nelun ) AEDU
+      AMZUBZUCZVAUDZCVANOZCENOZAVAUEACDUGMZVAADFNZCDUFMZNZCVEVAUHNZIACBVGJGUIVF
+      VHVICKPZLPZDUJMZUKDULMZQCVJVMQCVKVMQUMUNLVERKVERKLVEVMCDVLEFUSVESHUSSVMSV
+      LSUOTUPUQVBVCVDCUTNOVAEUTCURTUP $.
+  $}
+
+  ${
+    $d K x y $.  $d P x y $.  $d R x y $.  $d ph x y $.
+    rsprprmprmidl.k $e |- K = ( RSpan ` R ) $.
+    rsprprmprmidl.r $e |- ( ph -> R e. CRing ) $.
+    rsprprmprmidl.p $e |- ( ph -> P e. ( RPrime ` R ) ) $.
+    $( In a commutative ring, ideals generated by prime elements are prime
+       ideals.  (Contributed by Thierry Arnoux, 18-May-2025.) $)
+    rsprprmprmidl $p |- ( ph -> ( K ` { P } ) e. ( PrmIdeal ` R ) ) $=
+      ( vx vy ccrg wcel cfv wne eqid wbr wa adantr ellpi ad3antrrr biimpar wral
+      csn clidl cbs cv cmulr co wo cprmidl crg wss crngringd rprmcl snssd rspcl
+      wi crpm syl2anc cur wn ringidcl syl cdsr rprmnunit simpr dvdsunit syl3anc
+      cui 1unit mtbird nelne1 necomd wb ad2antrr simpllr simplr biimpa rprmdvds
+      mtand orim12da ex anasss ralrimivva w3a isprmidlc syl13anc ) ACJKZBUBZDLZ
+      CUCLZKZWICUDLZMZHUEZIUEZCUFLZUGZWIKZWNWIKZWOWIKZUHZUPZIWLUAHWLUAZWICUILKZ
+      FACUJKZWHWLUKWKACFULZABWLAWLCUQLZCJBWLNZXGNZFGUMZUNWLCWJWHDEXHWJNUOURAWLW
+      IACUSLZWLKZXKWIKZUTWLWIMAXEXLXFWLCXKXHXKNZVAVBAXMBXKCVCLZOZAXPBCVHLZKZAXG
+      BCXQJXIXQNZFGVDAXPPWGXPXKXQKZXRAWGXPFQAXPVEAXTXPAXEXTXFCXQXKXSXNVIVBQXOCX
+      QXKBXSXONZVFVGVSAWLXOCDBXKXHEYAXFXJRVJXKWLWIVKURVLAXBHIWLWLAWNWLKZWOWLKZX
+      BAYBPZYCPZWRXAYEWRPZBWNXOOZBWOXOOZWSWTYFWSYGAWSYGVMYBYCWRAWLXOCDBWNXHEYAX
+      FXJRSTYFWTYHYFWLXOCDBWOXHEYAYEXEWRAXEYBYCXFVNZQYEBWLKZWRAYJYBYCXJVNZQRTYF
+      WLXOXGBCWPJWNWOXHXIYAWPNZAWGYBYCWRFSABXGKYBYCWRGSAYBYCWRVOYDYCWRVPYEWRBWQ
+      XOOYEWLXOCDBWQXHEYAYIYKRVQVRVTWAWBWCWGXDWKWMXCWDHIWLWICWPXHYLWETWF $.
+  $}
+
+  ${
+    $d B x y $.  $d K x y $.  $d R x y $.  $d X x y $.  $d ph x y $.
+    rsprprmprmidlb.0 $e |- .0. = ( 0g ` R ) $.
+    rsprprmprmidlb.b $e |- B = ( Base ` R ) $.
+    rsprprmprmidlb.p $e |- P = ( RPrime ` R ) $.
+    rsprprmprmidlb.k $e |- K = ( RSpan ` R ) $.
+    rsprprmprmidlb.r $e |- ( ph -> R e. IDomn ) $.
+    rsprprmprmidlb.x $e |- ( ph -> X e. B ) $.
+    rsprprmprmidlb.1 $e |- ( ph -> X =/= .0. ) $.
+    $( In an integral domain, an ideal is generated by a prime elements is a
+       prime iff that elememt is prime.  (Contributed by Thierry Arnoux,
```

Comment:
"an ideal is generated by a prime elements is a prime iff that elememt is prime."
=> "an ideal generated by a single element is a prime iff that element is prime."

Reply:
Updated in https://github.com/metamath/set.mm/pull/4837/commits/3cb77ca77c03e28c1c877ddab4daed48dd91fb30